components/org.wso2.carbon.identity.verification.onfido.api/org.wso2.carbon.identity.verification.onfido.api.common/src/main/java/org/wso2/carbon/identity/verification/onfido/api/common/error/APIError.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.verification.onfido.api.common.error;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+/**
+ * Common Exception class for all the API related errors.
+ */
+public class APIError extends WebApplicationException {
+
+    private static final long serialVersionUID = -5096160501895108984L;
```

Comment:
Where is this being used?

Reply:
The `serialVersionUID` in the `APIError` class is necessary because it extends `WebApplicationException`, which implements `Serializable`. Even though the class might not be explicitly serialized right now, including `serialVersionUID` is a best practice. It ensures compatibility if the class is ever serialized in the future or if the class structure changes.

If `serialVersionUID` is excluded, Java will automatically generate one based on the class's details (e.g., fields, methods). However, this generated value is compiler-specific and may change if the class definition changes, potentially leading to deserialization issues.

By explicitly defining `serialVersionUID`, we prevent these issues and ensure smooth version control, making the code more robust and future-proof.